lib/xmlhasher/node.rb

```diff
@@ -11,19 +11,36 @@ def initialize(name)
     end
 
     def to_hash
-      h = {}
-      if text && !text.empty?
-        h[name] = text
-      else
-        h[name] = attributes.each_with_object({}) { |(key, value), r| r[key] = value if !value.nil? && !value.to_s.empty?; }
-        if children.size == 1
-          h[name].merge!(children.first.to_hash)
-        else
-          h[name].merge!(children.group_by(&:name).each_with_object({}) { |(k, v), r| v.length == 1 ? r.merge!(v.first.to_hash) : r[k] = v.map { |c| c.to_hash[c.name] }; })
+      node_content = content
+      { name => node_content.empty? ? nil : node_content }
+    end
+
+    private
+
+    def content
+      return text if text && !text.empty?
+
+      prepare_attributes.merge(prepare_children)
+    end
+
+    def prepare_attributes
```

Comment:
Maybe rename into `attributes_to_hash` and `children_to_hash`? @distroid what do you think?